extensions/common/iam/verifiable-credentials/src/main/java/org/eclipse/edc/iam/verifiablecredentials/rules/HasValidIssuer.java

```diff
@@ -42,6 +41,15 @@ public Result<Void> apply(VerifiableCredential credential) {
         if (issuer.id() == null) {
             return failure("Issuer did not contain an 'id' field.");
         }
-        return trustedIssuers.contains(issuer.id()) ? success() : failure("Issuer '%s' is not in the list of trusted issuers".formatted(issuer.id()));
+
+        if (!trustedIssuerRegistry.isTrusted(issuer)) {
```

Comment:
is this check needed? Simply checking whether a particular credential type is registered for the issuer seems enough. If the issuer is not registered, the registry would return an empty `Set`.